src/status_im/ui/screens/wallet/send/events.cljs

```diff
@@ -348,8 +351,10 @@
 
 (handlers/register-handler-fx
  :close-transaction-sent-screen
- (fn [{:keys [db]} _]
-   {:dispatch (if (= :wallet-send-transaction (second (:navigation-stack db)))
-                [:navigate-to-clean :wallet]
-                [:navigate-back])
+ (fn [{:keys [db]} [_ chat-id]]
+   {:dispatch-n (condp = (second (:navigation-stack db))
+                  :wallet-send-transaction [[:navigate-to-clean :wallet]]
+                  :wallet-send-transaction-chat [[:execute-stored-command]
```

Comment:
I don't like that we again start to degrade into unnecessary dispatch chains for encoding logic. We (painfully) mostly removed that from chat input/send handling and should take care to not re-introduce it in wallet.

Reply:
I don't like this one at all because of `:navigation-stack` lookup. Unfortunately, I'm stuck with this logic until I also change dapp transactions to use this same workflow. What I can do now is compose those two handlers from the second branch and replace `:dispatch-n` with `:dispatch`

Reply:
Awesome, that would be already much better, because at least, we would just retain the number of dispatches and not grow them :)